Framework/Kernel/inc/MantidKernel/V3D.h

```diff
@@ -331,6 +331,8 @@ class MANTID_KERNEL_DLL V3D final {
   void saveNexus(::NeXus::File *file, const std::string &name) const;
   void loadNexus(::NeXus::File *file, const std::string &name);
 
+  static bool isnan(V3D const &vec) { return (std::isnan(vec.X()) || std::isnan(vec.Y()) || std::isnan(vec.Z())); }
```

Comment:
Please add a doxygen string for this. It won't be clear without looking at the implementation that it is true if any of the elements are nans. IDEs will give it as a context when using the method with the doxygen.